compiler/src/main/java/com/squareup/anvil/compiler/codegen/dagger/ProvidesMethodFactoryGenerator.kt

```diff
@@ -102,9 +121,81 @@ internal class ProvidesMethodFactoryGenerator : PrivateCodeGenerator() {
         .withJvmSuppressWildcardsIfNeeded(function)
     val returnTypeIsNullable = function.typeReference?.isNullable() ?: false
 
+    return generateFactoryClass(
+        codeGenDir = codeGenDir,
+        clazz = clazz,
+        packageName = packageName,
+        className = className,
+        functionName = functionName,
+        isObject = isObject,
+        parameters = parameters,
+        returnType = returnType,
+        returnTypeIsNullable = returnTypeIsNullable,
+        isProperty = false
+    )
+  }
+
+  @OptIn(ExperimentalStdlibApi::class)
+  private fun generateFactoryClassForProperty(
+    codeGenDir: File,
+    module: ModuleDescriptor,
+    clazz: KtClassOrObject,
+    property: KtProperty
+  ): GeneratedFile {
+    val isCompanionObject = property.parents
+        .filterIsInstance<KtObjectDeclaration>()
+        .firstOrNull()
+        ?.isCompanion()
+        ?: false
+    val isObject = isCompanionObject || clazz is KtObjectDeclaration
+
+    val packageName = clazz.containingKtFile.packageFqName.asString()
+    val className = "${clazz.generateClassName()}_" +
+        (if (isCompanionObject) "Companion_" else "") +
+        "Get${property.requireFqName().shortName().asString().capitalize(US)}Factory"
+
+    val propertyName = property.nameAsSafeName.asString()
+
+    val returnType = property.requireTypeReference().requireTypeName(module)
+        .withJvmSuppressWildcardsIfNeeded(property)
+    val returnTypeIsNullable = property.typeReference?.isNullable() ?: false
+
+    return generateFactoryClass(
+        codeGenDir = codeGenDir,
+        clazz = clazz,
+        packageName = packageName,
+        className = className,
+        functionName = propertyName,
+        isObject = isObject,
+        parameters = emptyList(),
+        returnType = returnType,
+        returnTypeIsNullable = returnTypeIsNullable,
+        isProperty = true
+    )
+  }
+
+  @OptIn(ExperimentalStdlibApi::class)
+  private fun generateFactoryClass(
+    codeGenDir: File,
+    clazz: KtClassOrObject,
+    packageName: String,
+    className: String,
+    functionName: String,
```

Comment:
super-nit: no longer just a function name. Is there a word other than "statement" that describes both?

Reply:
Hm, the common name would be `callableDeclaration`. I'll still with `function` for now, because it makes it easier to read and understand the code for me. I'll add a comment, though.

Reply:
I've written that and immediately changed my mind. `declaration` it is.